torch/distributed/tensor/_ops/_view_ops.py

```diff
@@ -442,6 +452,21 @@ def dim_reduction(
     )
 
 
+def compute_shape(input_shape: Shape, output_rules: DimMap) -> Shape:
```

Comment:
what do folks think about this new util- is it worth adding, and OK with my approach of adding 'concrete_shape' fields to the different DimSpec impls?  I would complete the others too but did not do it yet.